### PR TITLE
Turno cibo

### PR DESCRIPTION
**Pull request by francescoluciano**
Now you can insert your Turno Cibo from Calendar

### CODE REVIEW

File: lib/model/allenamento.dart [hunk 1]
```diff
@@ -17,27 +19,35 @@ class Allenamento extends Evento {
...
   String toString() {
     return '$dataEOra $luogo $idSquadra $turnoCibo';
   }
 
   static Allenamento fromJson(String uid, Map<String, dynamic> json) {
     print('Data e ora: ${json["dataEOra"].toDate()}');
+    List<String> turni = [];
+    if (json['turnoCibo'] != null) {
+      json['turnoCibo'].forEach((t) => turni.add(t));
+    }
```
Comment (lucamerighi) [inline lib/model/allenamento.dart]: `json['turnoCibo']?.forEach((t) => turni.add(t));` should be enough

File: lib/model/squadra_partecipante.dart [hunk 2]
```diff
@@ -4,19 +4,25 @@ class SquadraPartecipante {
...
   static SquadraPartecipante fromJson(Map<String, dynamic> json) {
     List<Convocazione> conv = [];
-    json['convocazioni'].forEach((c) => conv.add(Convocazione.fromJson(c)));
+    List<String> turni = [];
+    if (json['convocazioni'] != null) {
+      json['convocazioni'].forEach((c) => conv.add(Convocazione.fromJson(c)));
+    }
+    if (json['turnoCibo'] != null) {
+      json['turnoCibo'].forEach((t) => turni.add(t));
+    }
```
Comment (lucamerighi) [inline lib/model/squadra_partecipante.dart]: same as above

File: lib/services/turnocibo_db.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import 'package:alley_app/model/evento.dart';
+import 'package:alley_app/model/partita.dart';
+import 'package:alley_app/model/user.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class TurnoCiboDbService {
+  final CollectionReference practiceCollection = Firestore.instance.collection('practices');
+  final CollectionReference gamesCollection = Firestore.instance.collection('games');
+
+  Future insertTurnoCibo(User user, Evento e) async {
```
Comment (lucamerighi) [inline lib/services/turnocibo_db.dart]: nothing to return, remove Future

File: lib/services/turnocibo_db.dart [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      casaOspite == 'casa' ? e.casa.turnoCibo.add(user.displayName) : e.ospite.turnoCibo.add(user.displayName);
+    } else {
+      practiceCollection.document(e.uid).updateData({
+        'turnoCibo': FieldValue.arrayUnion([user.displayName])
+      });
+      (e as Allenamento).turnoCibo.add(user.displayName);
+    }
+  }
+
+  Future removeTurnoCibo(User user, Evento e) async {
```
Comment (lucamerighi) [inline lib/services/turnocibo_db.dart]: nothing to return, remove Future

File: lib/views/informazioni/view_calendario.dart [hunk 5]
```diff
@@ -186,11 +189,8 @@ class _ViewCalendarioState extends State<ViewCalendario> with TickerProviderStat
         style: TextStyle(fontWeight: FontWeight.bold, fontSize: 20),
       ),
       SizedBox(height: 20),
-      RaisedButton(
-        color: Colors.orangeAccent,
-        onPressed: () => {print('hi')},
-        child: Text('Turno Cibo'),
-      )
+      Text('Turno cibo: ${p.getTurnoCibo(user.idSquadra).isNotEmpty ? p.getTurnoCibo(user.idSquadra) : "nessuno"}'),
```
Comment (lucamerighi) [inline lib/views/informazioni/view_calendario.dart]: double quotes

File: lib/views/informazioni/view_calendario.dart [hunk 6]
```diff
@@ -203,7 +203,27 @@ class _ViewCalendarioState extends State<ViewCalendario> with TickerProviderStat
         style: TextStyle(fontSize: 18),
       ),
       SizedBox(height: 20),
-      Text('Turno cibo: ${a.turnoCibo != '' ? a.turnoCibo : "nessuno"}'),
+      Text('Turno cibo: ${a.turnoCibo.isNotEmpty ? a.turnoCibo : "nessuno"}'),
```
Comment (lucamerighi) [inline lib/views/informazioni/view_calendario.dart]: double quotes

**Review by lucamerighi — CHANGES_REQUESTED** [review]

File: lib/views/informazioni/view_calendario.dart (hunk 5 above)
Comment (francescoluciano) [reply lib/views/informazioni/view_calendario.dart:192]: I can't, they're inside single quotes

**Review by lucamerighi — APPROVED** [review]

**Pull request merged**